CONTRIBUTING.md

```diff
@@ -45,22 +45,119 @@ To add a new plugin you need to do the following.
+| **maintainers**            | Maintainers for the plugin.                                                                                                                                                                                  | Yes      | N/A           |
+| **url**                    | The URL of the plugin's repository.                                                                                                                                                                          | Yes      | If not set, `package`'s' `homepage` attribute |
+| **package**                | The nixpkgs package attr for this plugin. Can be a string, a list of strings, a module option, or any derivation. For example, "foo-bar-nvim" for `pkgs.vimPlugins.foo-bar-nvim`, or `[ "hello" "world" ]` will be referenced as `pkgs.hello.world`. | No       | `name` parameter |
+| **imports**                | Additional Nix modules to import.                                                                                                                                                                            | No       | `[]`            |
+| **description**            | A brief description of the plugin.                                                                                                                                                                           | No       | `null`          |
+| **deprecateExtraOptions**  | Indicating whether to deprecate the `extraOptions` attribute. Mainly used for old plugins.                                                                                                                   | No       | `false`         |
+| **optionsRenamedToSettings** | Options that have been renamed and move to the `settings` attribute.                                                                                                                                       | No       | `[]`            |
+| **isColorscheme**          | Indicating whether the plugin is a colorscheme.                                                                                                                                                              | No       | `false`         |
+| **colorscheme**            | The name of the colorscheme.                                                                                                                                                                                 | No       | `name` parameter |
+| **configLocation**         | The location for the Lua configuration.                                                                                                                                                                      | No       | `"extraConfigLuaPre"` if `isColorscheme` is `true`, otherwise `"extraConfigLua"` |
```

Comment:
```suggestion
| **configLocation**         | The location for the Lua configuration.                                                                                                                                                                      | No       | if `isColorscheme` then `"extraConfigLuaPre"`, otherwise `"extraConfigLua"` |
```